### PR TITLE
Benchmark use cases

### PR DESCRIPTION
**Pull request by samuelsenior**
Added some use case compositions to the benchmarks, added the last-benches.toml file for sorting purposes, and removed its exclusion from .gitignore. Also specified a newer version of tokio in monument's Cargo.toml file to fix a compilation error.

### CODE REVIEW

File: .gitignore
Comment (kneasle) [inline .gitignore:7]: I think we should rename the current `.last-benches.toml` file rather than removing this from the `.gitignore`.  The `.last-benches.toml` is overwritten every time the benches get run, so tracking it with git is likely to cause some horrific merge conflicts...

We basically just want another file to enforce a rough ordering on the benchmarks, so we should probably make it completely different to the others (which have their own purposes).  I'm not sure what to name this file, though, maybe a non-hidden file with a name like `baseline-benches.toml`?

**Review by kneasle — CHANGES_REQUESTED** [review]
Very nice!  I like the benchmarks, just two tweaks from me before merging :)

**Comment by samuelsenior** [comment]
I think the changes made should address the tweaks asked for. If they don't, or if you see anything else, just say!

**Comment by kneasle** [comment]
Excellent, lovely stuff!  I'll merge as soon as the automated checks finish (they'll almost certainly pass because the code hasn't changed).

**Review by kneasle — APPROVED** [review]
Looks good to me (apparently I have to explicitly make another review before GitHub is happy to merge)...

**Comment by kneasle** [comment]
One last thing before merging - would it be possible to rebase this branch onto `kneasle/master` rather than merging it?  There's a general rule of thumb to rebase out of the main branch and merge into the main branch, which leaves a very clean commit history without lots of criss-crossing merges.

I don't think there should be merge conflicts because you're modifying benchmarks and the changes to `master` have been to `cli/` and `lib/`.

**Comment by samuelsenior** [comment]
Ah, sorry, I didn't realise! I think it should now be rebased onto kneasle/master, though if I missed something just say!

**Pull request merged**
